src/DwellingAPI.php

```diff
@@ -39,36 +39,15 @@ public function checkIsUp(): bool
         }
         return false;
     }
+
     /**
      * Load the dwellings in from the API
      * 
      * @return Array an indexed array of objects. Each object represents a dwelling
      */
-    public function loadDwellings() : Array
-    {
-        curl_setopt($this->apiConnection, CURLOPT_URL, $this->apiUrl . 'properties.json');
-        $output = curl_exec($this->apiConnection);
-        return json_decode($output);
-    }
-    /**
-     * Load the types in from the API
-     * 
-     * @return Array an indexed array of objects. Each object represents a type
-     */
-    public function loadTypes() : Array
-    {
-        curl_setopt($this->apiConnection, CURLOPT_URL, $this->apiUrl . 'types.json');
-        $output = curl_exec($this->apiConnection);
-        return json_decode($output);
-    }
-    /**
-     * Load the statuses in from the API
-     * 
-     * @return Array an indexed array of objects. Each object represents a status
-     */
-    public function loadStatuses() : Array
+    public function loadEndpoint(string $anEndpoint) : Array
```

Comment:
This is better. It might be worth adding back the original methods of loadProperties/types/statuses and then making them call the loadEndpoint method (which could then be private).

That way you don't have to remember what the actual endpoint is when you need to use the API, you can just directly call one of the load methods, and it's slightly less open to abuse because you can't directly access the loadEndpoint method and give it something dodgy.